src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -135,6 +135,7 @@ namespace cryptonote
     std::atomic<bool> m_synchronized;
     bool m_one_request = true;
     std::atomic<bool> m_stopping;
+    epee::critical_section m_sync_lock;
```

Comment:
Ok, consistency was my only guess.

The [commit that added the sleep](https://github.com/monero-project/monero/commit/ae2a50659f7dc74a5446a0dc3a5f8f78563b9e1f). Based on the information in that commit, it appears the sleep was for debugging purposes. Since the codebase has not consistently used epee locking, the runtime option is not consistently applying the sleep. Luckily when default of zero is used, `boost::thread` will return immediately. But it forces a read from static memory before returning.

Have you (or @hyc @luigi1111 @tewinget @iamsmooth @anonimal et all ) ever debugged multi-threaded code in this manner? Maybe the original author needs to be contacted. I am trying to think of situation this would help - maybe if there was some racy code after the lock was released? But then every thread is forced to sleep the same amount of time, so ... ?